### ISSUE TITLE
[gateway] Authentication issues with Windows Server 2012 R2

### ISSUE THREAD

**Issue by Olen** [FreeRDP/FreeRDP#5069]
Opening a new issue, ref #5022 

**Describe the bug**
When trying to connect to a farm of windows-servers, I get an ERRCONNECT_CONNECT_TRANSPORT_FAILED

Unfortunately, I am in no control over the servers, but they are running Windows Server 2012 R2.

I can connect using freerdp-2.0.0-41.20180405gita9ecd6a.fc28.x86_64.rpm
But no available build after that is able to connect (due to different issues discussed in various cases).

Command:
`/opt/freerdp-nightly/bin/xfreerdp $HOME/.config/rdp/cpub-FARM10210_Smart-FARM10210_Smart-CmsRdsh.rdp /smart-sizing:2560x1440 /size:2560x1440 /u:user@domain /grab-keyboard +home-drive /log-level:trace -sec-ext +sec-nla -sec-tls -sec-rdp`

Content of the rdp-file:
```
redirectclipboard:i:1
redirectprinters:i:1
redirectcomports:i:1
redirectsmartcards:i:1
devicestoredirect:s:*
drivestoredirect:s:*
redirectdrives:i:1
session bpp:i:32
prompt for credentials on client:i:1
server port:i:3389
allow font smoothing:i:1
promptcredentialonce:i:1
videoplaybackmode:i:1
audiocapturemode:i:0
gatewayusagemethod:i:1
gatewayprofileusagemethod:i:1
gatewaycredentialssource:i:0
full address:s:RDCB01.DOORWAY.NO
gatewayhostname:s:rd.doorway.no
workspace id:s:RDCB01.DOORWAY.NO
use redirection server name:i:1
loadbalanceinfo:s:tsv://MS Terminal Services Plugin.1.FARM10210_Smart
use multimon:i:0
alternate full address:s:RDCB01.DOORWAY.NO
signscope:s:Full Address,Alternate Full Address,Use Redirection Server Name,Server Port,GatewayHostname,GatewayUsageMethod,GatewayProfileUsageMethod,GatewayCredentialsSource,PromptCredentialOnce,RedirectDrives,RedirectPrinters,RedirectCOMPorts,RedirectSmartCards,RedirectClipboard,DevicesToRedirect,DrivesToRedirect,LoadBalanceInfo
signature:s:AQABAAEAAADvDwAAMIIP.........
```

The session ends with:
```
[13:40:40:506] [19190:19193] [DEBUG][com.freerdp.core.nla] - CredSSP protocol support 6, peer supports 6
[13:40:40:506] [19190:19193] [TRACE][com.freerdp.core.nla] - InitializeSecurityContext  SEC_I_COMPLETE_NEEDED [0x00090313]
[13:40:40:506] [19190:19193] [ERROR][com.winpr.sspi] - EncryptMessage status SEC_E_INVALID_TOKEN [0x80090308]
[13:40:40:506] [19190:19193] [ERROR][com.freerdp.core.nla] - EncryptMessage status SEC_E_INVALID_TOKEN [0x80090308]
[13:40:40:506] [19190:19193] [ERROR][com.freerdp.core.rdp] - rdp_recv_callback: CONNECTION_STATE_NLA - nla_recv_pdu() fail
[13:40:40:506] [19190:19193] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
[13:40:40:506] [19190:19193] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[13:40:40:506] [19190:19193] [ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
```

Here it hangs for 1:30 minutes while consuming a lot of CPU, before failing:

```
[13:42:10:570] [19190:19193] [ERROR][com.freerdp.core.transport] - BIO_read returned a system error 104: Connection reset by peer
[13:42:10:571] [19190:19193] [DEBUG][com.freerdp.core.transport] - transport_check_fds: transport_read_pdu() - -1
[13:42:10:571] [19190:19193] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[13:42:10:604] [19190:19193] [DEBUG][com.freerdp.core.nego] - Enabling security layer negotiation: TRUE
[13:42:10:604] [19190:19193] [DEBUG][com.freerdp.core.nego] - Enabling restricted admin mode: FALSE
[13:42:10:604] [19190:19193] [DEBUG][com.freerdp.core.nego] - Enabling RDP security: FALSE
[13:42:10:604] [19190:19193] [DEBUG][com.freerdp.core.nego] - Enabling TLS security: FALSE
[13:42:10:604] [19190:19193] [DEBUG][com.freerdp.core.nego] - Enabling NLA security: TRUE
[13:42:10:604] [19190:19193] [DEBUG][com.freerdp.core.nego] - Enabling NLA extended security: FALSE
[13:42:10:604] [19190:19193] [DEBUG][com.freerdp.core.nego] - state: NEGO_STATE_NLA
[13:42:10:604] [19190:19193] [DEBUG][com.freerdp.core.nego] - Attempting NLA security
[13:42:10:618] [19190:19193] [DEBUG][com.freerdp.core] - connecting to peer 185.220.208.3
[13:42:10:638] [19190:19193] [DEBUG][com.winpr.sspi] - InitSecurityInterfaceExA
[13:42:10:639] [19190:19193] [TRACE][com.freerdp.core.gateway.ntlm] - InitializeSecurityContext status SEC_I_CONTINUE_NEEDED [0x00090312]
[13:42:10:645] [19190:19193] [TRACE][com.freerdp.core.gateway.ntlm] - InitializeSecurityContext status SEC_I_COMPLETE_NEEDED [0x00090313]
[13:42:10:653] [19190:19193] [DEBUG][com.freerdp.core.gateway.rdg] - RDG_OUT_DATA authorization result: 200
[13:42:10:653] [19190:19193] [DEBUG][com.freerdp.core] - connecting to peer 185.220.208.3
[13:42:10:670] [19190:19193] [DEBUG][com.winpr.sspi] - InitSecurityInterfaceExA
[13:42:10:670] [19190:19193] [TRACE][com.freerdp.core.gateway.ntlm] - InitializeSecurityContext status SEC_I_CONTINUE_NEEDED [0x00090312]
[13:42:10:674] [19190:19193] [TRACE][com.freerdp.core.gateway.ntlm] - InitializeSecurityContext status SEC_I_COMPLETE_NEEDED [0x00090313]
[13:42:10:684] [19190:19193] [DEBUG][com.freerdp.core.gateway.rdg] - RDG_IN_DATA authorization result: 200
[13:42:10:688] [19190:19193] [DEBUG][com.freerdp.core.gateway.rdg] - Handshake response received
[13:42:10:688] [19190:19193] [DEBUG][com.freerdp.core.gateway.rdg] - errorCode=RPC_S_OK [0x00000000], verMajor=1, verMinor=0, serverVersion=0, extendedAuth=HTTP_EXTENDED_AUTH_NONE [0x0000]
[13:42:10:692] [19190:19193] [DEBUG][com.freerdp.core.gateway.rdg] - Tunnel response received
[13:42:10:692] [19190:19193] [DEBUG][com.freerdp.core.gateway.rdg] - serverVersion=0, errorCode=RPC_S_OK [0x00000000], fieldsPresent=HTTP_TUNNEL_RESPONSE_FIELD_TUNNEL_ID|HTTP_TUNNEL_RESPONSE_FIELD_CAPS|HTTP_TUNNEL_RESPONSE_FIELD_SOH_REQ| [0007]|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_REDIR_FLAGS|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_IDLE_TIMEOUT| [0003]|HTTP_CHANNEL_RESPONSE_FIELD_CHANNELID|HTTP_CHANNEL_RESPONSE_OPTIONAL|HTTP_CHANNEL_RESPONSE_FIELD_UDPPORT [0007]|HTTP_TUNNEL_RESPONSE_FIELD_TUNNEL_ID|HTTP_TUNNEL_RESPONSE_FIELD_CAPS|HTTP_TUNNEL_RESPONSE_FIELD_SOH_REQ| [0007]|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_REDIR_FLAGS|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_IDLE_TIMEOUT| [0003]|HTTP_CHANNEL_RESPONSE_FIELD_CHANNELID|HTTP_CHANNEL_RESPONSE_OPTIONAL|HTTP_CHANNEL_RESPONSE_FIELD_UDPPORT [0007]|HTTP_TUNNEL_RESPONSE_FIELD_TUNNEL_ID|HTTP_TUNNEL_RESPONSE_FIELD_CAPS|HTTP_TUNNEL_RESPONSE_FIELD_SOH_REQ| [0007]
[13:42:10:706] [19190:19193] [DEBUG][com.freerdp.core.gateway.rdg] - Tunnel authorization received
[13:42:10:706] [19190:19193] [DEBUG][com.freerdp.core.gateway.rdg] - errorCode=RPC_S_OK [0x00000000], fieldsPresent=HTTP_TUNNEL_RESPONSE_FIELD_TUNNEL_ID|HTTP_TUNNEL_RESPONSE_FIELD_CAPS|HTTP_TUNNEL_RESPONSE_FIELD_SOH_REQ| [0007]|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_REDIR_FLAGS|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_IDLE_TIMEOUT| [0003]|HTTP_CHANNEL_RESPONSE_FIELD_CHANNELID|HTTP_CHANNEL_RESPONSE_OPTIONAL|HTTP_CHANNEL_RESPONSE_FIELD_UDPPORT [0007]|HTTP_TUNNEL_RESPONSE_FIELD_TUNNEL_ID|HTTP_TUNNEL_RESPONSE_FIELD_CAPS|HTTP_TUNNEL_RESPONSE_FIELD_SOH_REQ| [0007]|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_REDIR_FLAGS|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_IDLE_TIMEOUT| [0003]|HTTP_CHANNEL_RESPONSE_FIELD_CHANNELID|HTTP_CHANNEL_RESPONSE_OPTIONAL|HTTP_CHANNEL_RESPONSE_FIELD_UDPPORT [0007]|HTTP_TUNNEL_RESPONSE_FIELD_TUNNEL_ID|HTTP_TUNNEL_RESPONSE_FIELD_CAPS|HTTP_TUNNEL_RESPONSE_FIELD_SOH_REQ| [0007]|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_REDIR_FLAGS|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_IDLE_TIMEOUT| [0003]
[13:42:10:730] [19190:19193] [DEBUG][com.freerdp.core.gateway.rdg] - Channel response received
[13:42:10:730] [19190:19193] [DEBUG][com.freerdp.core.gateway.rdg] - channel response errorCode=RPC_S_OK [0x00000000], fieldsPresent=HTTP_TUNNEL_RESPONSE_FIELD_TUNNEL_ID|HTTP_TUNNEL_RESPONSE_FIELD_CAPS|HTTP_TUNNEL_RESPONSE_FIELD_SOH_REQ| [0007]|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_REDIR_FLAGS|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_IDLE_TIMEOUT| [0003]|HTTP_CHANNEL_RESPONSE_FIELD_CHANNELID|HTTP_CHANNEL_RESPONSE_OPTIONAL|HTTP_CHANNEL_RESPONSE_FIELD_UDPPORT [0007]|HTTP_TUNNEL_RESPONSE_FIELD_TUNNEL_ID|HTTP_TUNNEL_RESPONSE_FIELD_CAPS|HTTP_TUNNEL_RESPONSE_FIELD_SOH_REQ| [0007]|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_REDIR_FLAGS|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_IDLE_TIMEOUT| [0003]|HTTP_CHANNEL_RESPONSE_FIELD_CHANNELID|HTTP_CHANNEL_RESPONSE_OPTIONAL|HTTP_CHANNEL_RESPONSE_FIELD_UDPPORT [0007]|HTTP_TUNNEL_RESPONSE_FIELD_TUNNEL_ID|HTTP_TUNNEL_RESPONSE_FIELD_CAPS|HTTP_TUNNEL_RESPONSE_FIELD_SOH_REQ| [0007]|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_REDIR_FLAGS|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_IDLE_TIMEOUT| [0003]|HTTP_CHANNEL_RESPONSE_FIELD_CHANNELID|HTTP_CHANNEL_RESPONSE_OPTIONAL|HTTP_CHANNEL_RESPONSE_FIELD_UDPPORT [0007]
[13:42:10:730] [19190:19193] [DEBUG][com.freerdp.core.nego] - RequestedProtocols: 3
[13:42:10:755] [19190:19193] [DEBUG][com.freerdp.core.nego] - RDP_NEG_RSP
[13:42:10:755] [19190:19193] [DEBUG][com.freerdp.core.nego] - selected_protocol: 2
[13:42:10:755] [19190:19193] [DEBUG][com.freerdp.core.nego] - state: NEGO_STATE_FINAL
[13:42:10:755] [19190:19193] [DEBUG][com.freerdp.core.nego] - Negotiated NLA security
[13:42:10:755] [19190:19193] [DEBUG][com.freerdp.core.nego] - nego_security_connect with PROTOCOL_NLA
[13:42:10:768] [19190:19193] [DEBUG][com.winpr.sspi] - InitSecurityInterfaceExA
[13:42:10:768] [19190:19193] [DEBUG][com.freerdp.core.nla] - nla_client_init 349 : packageName=Kerberos ; cbMaxToken=48000
[13:42:10:768] [19190:19193] [TRACE][com.freerdp.core.nla] -  InitializeSecurityContext status SEC_I_CONTINUE_NEEDED [0x00090312]
[13:42:10:768] [19190:19193] [DEBUG][com.freerdp.core.nla] - Sending Authentication Token
[13:42:10:768] [19190:19193] [DEBUG][com.freerdp.core.nla] - 0000 4e 54 4c 4d 53 53 50 00 01 00 00 00 b7 82 08 e2 NTLMSSP.........
[13:42:10:768] [19190:19193] [DEBUG][com.freerdp.core.nla] - 0010 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 ................
[13:42:10:768] [19190:19193] [DEBUG][com.freerdp.core.nla] - 0020 06 01 b1 1d 00 00 00 0f                         ........
[13:42:10:773] [19190:19193] [DEBUG][com.freerdp.core.nla] - CredSSP protocol support 6, peer supports 6
[13:42:10:773] [19190:19193] [TRACE][com.freerdp.core.nla] - InitializeSecurityContext  SEC_I_COMPLETE_NEEDED [0x00090313]
[13:42:10:773] [19190:19193] [ERROR][com.winpr.sspi] - EncryptMessage status SEC_E_INVALID_TOKEN [0x80090308]
[13:42:10:773] [19190:19193] [ERROR][com.freerdp.core.nla] - EncryptMessage status SEC_E_INVALID_TOKEN [0x80090308]
[13:42:10:773] [19190:19193] [ERROR][com.freerdp.core.rdp] - rdp_recv_callback: CONNECTION_STATE_NLA - nla_recv_pdu() fail
[13:42:10:773] [19190:19193] [ERROR][com.freerdp.core.transport] - transport_check_fds: transport->ReceiveCallback() - -1
[13:42:10:773] [19190:19193] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
[13:42:10:773] [19190:19193] [ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_CONNECT_TRANSPORT_FAILED [0x0002000D]
[13:42:10:773] [19190:19193] [ERROR][com.freerdp.core] - freerdp_post_connect failed
```

**Comment by akallabeth** [comment]
Just checked, might be an issue with `use redirection server name` which is ignored. Previously the redirection was always using the `IP` address provided, but now it tries to use the `FQDN` if it is available.
Could you check, if it works with `/redirect-prefer:ip` or something similar?

**Comment by akallabeth** [comment]
Another option could be that the username/credentials for the gateway are not properly set, could you try with `/bu:<user> /gd:<domain> /gp:<pwd>` explicitly set?

**Comment by Olen** [comment]
I assume you meant /gu: (not /bu:)

With only `/redirect-prefer:ip` the 1:30 wait is gone (but it it still not able to connect).
With /gu: /gd: and /gp: it fails much earlier in the connection:

```
[14:32:23:941] [30438:30439] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpdr
[14:32:23:941] [30438:30439] [INFO][com.freerdp.client.common.cmdline] - loading channelEx rdpsnd
[14:32:23:941] [30438:30439] [DEBUG][com.freerdp.channels.cliprdr.client] - VirtualChannelEntryEx
[14:32:23:941] [30438:30439] [INFO][com.freerdp.client.common.cmdline] - loading channelEx cliprdr
[14:32:23:941] [30438:30439] [DEBUG][com.freerdp.client.x11] - Searching for XInput pointer device
[14:32:23:942] [30438:30439] [DEBUG][com.freerdp.client.x11] - Pointer device: 9
[14:32:23:944] [30438:30439] [DEBUG][com.freerdp.core.nego] - Enabling security layer negotiation: TRUE
[14:32:23:944] [30438:30439] [DEBUG][com.freerdp.core.nego] - Enabling restricted admin mode: FALSE
[14:32:23:944] [30438:30439] [DEBUG][com.freerdp.core.nego] - Enabling RDP security: FALSE
[14:32:23:944] [30438:30439] [DEBUG][com.freerdp.core.nego] - Enabling TLS security: FALSE
[14:32:23:944] [30438:30439] [DEBUG][com.freerdp.core.nego] - Enabling NLA security: TRUE
[14:32:23:944] [30438:30439] [DEBUG][com.freerdp.core.nego] - Enabling NLA extended security: FALSE
[14:32:23:944] [30438:30439] [DEBUG][com.freerdp.core.nego] - state: NEGO_STATE_NLA
[14:32:23:944] [30438:30439] [DEBUG][com.freerdp.core.nego] - Attempting NLA security
[14:32:23:956] [30438:30439] [DEBUG][com.freerdp.core] - connecting to peer 185.220.208.3
[14:32:23:010] [30438:30439] [DEBUG][com.winpr.sspi] - InitSecurityInterfaceExA
[14:32:23:010] [30438:30439] [TRACE][com.freerdp.core.gateway.ntlm] - InitializeSecurityContext status SEC_I_CONTINUE_NEEDED [0x00090312]
[14:32:23:016] [30438:30439] [TRACE][com.freerdp.core.gateway.ntlm] - InitializeSecurityContext status SEC_I_COMPLETE_NEEDED [0x00090313]
[14:32:23:022] [30438:30439] [DEBUG][com.freerdp.core.gateway.rdg] - RDG_OUT_DATA authorization result: 401
[14:32:23:023] [30438:30439] [ERROR][com.freerdp.core.nego] - Protocol Security Negotiation Failure
[14:32:23:023] [30438:30439] [ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_SECURITY_NEGO_CONNECT_FAILED [0x0002000C]
[14:32:23:023] [30438:30439] [ERROR][com.freerdp.core.connection] - Error: protocol security negotiation or connection failure
```

Removing the domain `/gd:` (but still keeping username and password) brings me back to the same error as before:

[14:37:33:815] [31755:31756] [ERROR][com.freerdp.core.nego] - Protocol Security Negotiation Failure
[14:37:33:815] [31755:31756] [ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_SECURITY_NEGO_CONNECT_FAILED [0x0002000C]
[14:37:33:815] [31755:31756] [ERROR][com.freerdp.core.connection] - Error: protocol security negotiation or connection failure
[14:37:33:815] [31755:31756] [DEBUG][com.freerdp.core.transport] - transport_check_fds: transport_read_pdu() - -1
[14:37:33:815] [31755:31756] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1

**Comment by akallabeth** [comment]
@Olen Very stupid question, you're using some kerberos setup locally?
If so, could you recompile the nightyly with `-DWITH_GSSAPI=OFF` and retest? (kerberos support is very basic at the moment)

**Comment by Olen** [comment]
Unfortunately no.  No special auth locally.

**Comment by akallabeth** [comment]
@Olen Looks like your issue is something quite different, could you give any additional details you can get a hold of?
Don't see an issue with the gateway protocol, we're now down to a case of authentication failure, maybe it is a specific setting that triggers this behaviour (but sadly there are many of these and we have to narrow it down first)

**Comment by Olen** [comment]
Updated to the latest nightly again

And I see this as the last error:
```
[11:50:35:488] [32456:32457] [DEBUG][com.freerdp.core.gateway.rdg] - channel response errorCode=**E_PROXY_RAP_ACCESSDENIED** [0x800759DA], fieldsPresent=HTTP_TUNNEL_RESPONSE_FIELD_TUNNEL_ID|HTTP_TUNNEL_RESPONSE_FIELD_CAPS|HTTP_TUNNEL_RESPONSE_FIELD_SOH_REQ| [0007]|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_REDIR_FLAGS|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_IDLE_TIMEOUT| [0003]|HTTP_CHANNEL_RESPONSE_FIELD_CHANNELID|HTTP_CHANNEL_RESPONSE_OPTIONAL|HTTP_CHANNEL_RESPONSE_FIELD_UDPPORT [0007]|HTTP_TUNNEL_RESPONSE_FIELD_TUNNEL_ID|HTTP_TUNNEL_RESPONSE_FIELD_CAPS|HTTP_TUNNEL_RESPONSE_FIELD_SOH_REQ| [0007]|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_REDIR_FLAGS|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_IDLE_TIMEOUT| [0003]||| [0000]
[11:50:35:488] [32456:32457] [ERROR][com.freerdp.core.gateway.rdg] - channel response errorCode=E_PROXY_RAP_ACCESSDENIED [0x800759DA], fieldsPresent=HTTP_TUNNEL_RESPONSE_FIELD_TUNNEL_ID|HTTP_TUNNEL_RESPONSE_FIELD_CAPS|HTTP_TUNNEL_RESPONSE_FIELD_SOH_REQ| [0007]|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_REDIR_FLAGS|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_IDLE_TIMEOUT| [0003]|HTTP_CHANNEL_RESPONSE_FIELD_CHANNELID|HTTP_CHANNEL_RESPONSE_OPTIONAL|HTTP_CHANNEL_RESPONSE_FIELD_UDPPORT [0007]|HTTP_TUNNEL_RESPONSE_FIELD_TUNNEL_ID|HTTP_TUNNEL_RESPONSE_FIELD_CAPS|HTTP_TUNNEL_RESPONSE_FIELD_SOH_REQ| [0007]|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_REDIR_FLAGS|HTTP_TUNNEL_AUTH_RESPONSE_FIELD_IDLE_TIMEOUT| [0003]||| [0000]||| [0000]
[11:50:35:488] [32456:32457] [ERROR][com.freerdp.core.nego] - Protocol Security Negotiation Failure
[11:50:35:488] [32456:32457] [ERROR][com.freerdp.core] - freerdp_set_last_error ERRCONNECT_SECURITY_NEGO_CONNECT_FAILED [0x0002000C]
[11:50:35:488] [32456:32457] [ERROR][com.freerdp.core.connection] - Error: protocol security negotiation or connection failure
[11:50:35:488] [32456:32457] [DEBUG][com.freerdp.core.transport] - transport_check_fds: transport_read_pdu() - -1
[11:50:35:488] [32456:32457] [DEBUG][com.freerdp.core.rdp] - transport_check_fds() - -1
```

This is FreeRDP version 2.0.0-dev5 (e7a44bc43)

So it should include the fix that made it work for @gsacre in #4663 

I am also available on irc, if it makes conversation and testing different options easier.

**Comment by akallabeth** [comment]
@Olen Ok, `E_PROXY_RAP_ACCESSDENIED` means `Returned when an attempt to resolve or access a
target server is blocked by RDG server policies.`
Might be a good idea to not use the rdp file and just try to connect with a command line to find out which setting is at fault here.

**Comment by akallabeth** [comment]
As discussed on IRC looks like issues with kerberos/gss.
Closing this one, tracking with #4311

**Comment by Olen** [comment]
For others, running into the same (or similar) problems.

I had to rebuild the RPM and disable GSSAPI:
Change packaging/rpm/freerdp-nightly.spec and set
-DWITH_GSSAPI=OFF
Under cmake-flags

And then create a new rpm (rpmbuild -bb)

**Comment by akallabeth** [comment]
@ondrejholy you know who to contact for the fedora rpm? might be a good idea to deactivate GSSAPI/kerberos support for the time being as it is breaking more that fixing.

**Comment by ondrejholy** [comment]
@akallabeth It is currently disabled: https://src.fedoraproject.org/rpms/freerdp/c/a876205f1737ecbb5a6f71e3db2da64bccbb3eca?branch=master. You can find the maintainers and other info on https://src.fedoraproject.org/rpms/freerdp. You can also submit pull requests there. However, I've submitted some requests during the last weeks and they have left unnoticed. I will try to obtain the commit rights for this package...
